modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/GetLinterCustomRulesThroughThePublisherRestAPITestCase.java

```diff
@@ -0,0 +1,104 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ *
+ */
+
+package org.wso2.am.integration.tests.publisher;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.testng.annotations.*;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.APIListDTO;
+import org.wso2.am.integration.test.utils.base.APIMIntegrationBaseTest;
+import org.wso2.am.integration.test.utils.bean.APIRequest;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+
+import javax.ws.rs.core.Response;
+import java.io.File;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Get linter custom rules through the publisher REST API
+ * APIM-534 / APIM-542
+ */
+
+public class GetLinterCustomRulesThroughThePublisherRestAPITestCase extends
+        APIMIntegrationBaseTest {
+
+    private static final Log log = LogFactory.
+            getLog(GetLinterCustomRulesThroughThePublisherRestAPITestCase.class);
+    private JSONObject tenantConfig;
+    JSONParser jsonParser = new JSONParser();
+    String linterCustomRulesKey = "LinterCustomRules";
+
+    @Factory(dataProvider = "userModeDataProvider")
+    public GetLinterCustomRulesThroughThePublisherRestAPITestCase(TestUserMode userMode) {
+
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+
+        return new Object[][]{
+                new Object[]{TestUserMode.SUPER_TENANT_ADMIN},
+//                new Object[]{TestUserMode.TENANT_ADMIN},
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+
+        super.init(userMode);
+        String tenantConfContent = FileUtils.readFileToString(new File(getAMResourceLocation() + File.separator
+                + "configFiles" + File.separator + "linterCustomRulesTest" + File.separator + "tenant-conf.json"), "UTF-8");
+        tenantConfig = (JSONObject) jsonParser.parse(tenantConfContent);
+        restAPIAdmin.updateTenantConfig(tenantConfig);
+
+    }
+
+    @Test(description = "Create an API using valid data and get all " +
+            "the API through Publisher Rest API")
```

Comment:
Shall we change the test description to match the actual test case ? (ex: Test get linter rules through publisher REST API)

Reply:
Change added in [dce5303](https://github.com/wso2/product-apim/pull/12971/commits/dce5303624fbab0df523d5d4a63a9f2152868a8b)